docs/best_practices/extensions.rst

```diff
@@ -16,15 +16,104 @@ If an extension is required, tutorials for the process may be found through the
+                                         doc='Included group of ype MyNewType1')]
+                )
+
+            .. code-tab:: yaml YAML
+
+                groups:
+                - neurodata_type_def: MyNewType1
+                  neurodata_type_inc: LabMetaData
+                  name: custom_type1
+                  doc: Example extension type 1
+                - neurodata_type_def: MyNewType2
+                  neurodata_type_inc: NWBContainer
+                  name: custom_type2
+                  doc: Example extension type 2
+                  groups:
+                  - neurodata_type_inc: MyNewType1
+                    doc: Included group of ype MyNewType1
+
+    .. tab:: DON'T do this
+
+        .. tabs::
+
+            .. code-tab:: py Python
+
+                from pynwb.spec import NWBGroupSpec
+
+                # Do NOT define a new type via ``neurodata_type_def`` within the
+                # definition of another type. Always define the types separately
+                # and use ``neurodata_type_inc`` to include the type
+                type2_ext = NWBGroupSpec(
+                    name='custom_type2',
+                    doc='Example extension type 2',
+                    neurodata_type_def='MyNewType2',
+                    groups=[NWBGroupSpec(
+                        name='custom_type1',
+                        doc='Example extension type 1',
+                        neurodata_type_def='MyNewType1',
+                        neurodata_type_inc='LabMetaData',
+                    )]
+                )
+
+            .. code-tab:: yaml YAML
+
+                groups:
+                - neurodata_type_def: MyNewType2
+                  neurodata_type_inc: NWBContainer
+                  name: custom_type2
+                  doc: Example extension type 2
+                  groups:
+                  - neurodata_type_def: MyNewType1
+                    neurodata_type_inc: LabMetaData
+                    name: custom_type1
+                    doc: Example extension type 1
+
+Build on and Reuse Existing Neurodata Types
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 When possible, use existing types when creating extensions either by creating new ``neurodata_types`` that inherit from
 existing ones, or by creating ``neurodata_types`` that contain existing ones. Building on existing types facilitates the
 reuse of existing functionality and interpretation of the data. If a community extension already exists that has a
-similar scope, it is preferable to use that extension rather than creating a new one.
+similar scope, it is preferable to use that extension rather than creating a new one. For example:
+
+* Extend ``TimeSeries`` for storing timeseries data. NWB provides main types of ``TimeSeries``
+  and you should identify the most specific type of ``TimeSeries`` relevant for your use case
+  (e.g., extend ``ElectricalSeries`` to define a new kind of electrical recording).
```

Comment:
Should these `TimeSeries`/`ElectricalSeries` be :py:class: intersphinx references?

Reply:
Ditto for below on `TimeIntervals` / `DynamicTable`

Reply:
I would suggest intersphinx to the nwb-schema docs since this is for extensions